react/src/components/NavBar/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+import 'bootstrap/dist/css/bootstrap.css';
+
+import UserAvatar from '../UserAvatar';
+import routes from '../../constants/routes'
+import wbooksLogoSvg from '../../assets/wbooks_logo.svg';
+import addBookSvg from '../../assets/add_book.svg';
+import notificationsSvg from '../../assets/notifications.svg';
+
+import './styles.css';
+
+class NavBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      dropdownOpen: false
+    };
+  }
+
+  toggle() {
+    this.setState({
+      dropdownOpen: !this.state.dropdownOpen
+    });
+  }
+  render() {
+    return (
+      <nav className="navbar">
+        <NavLink to={routes.HOME()}>
+          <img className="wbooks-logo" src={wbooksLogoSvg} alt="Wbooks" />
+        </NavLink>
+        <div className="navbar-icons-group">
+          <img src={notificationsSvg} alt="Wolox" />
+          <img src={addBookSvg} alt="Wolox" />
+          <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
+            <DropdownToggle
+              tag="div"
+              onClick={this.toggle}
+              data-toggle="dropdown"
+              aria-expanded={this.state.dropdownOpen}>
+              <UserAvatar />
+            </DropdownToggle>
+            <DropdownMenu
+              right={true}
+            >
+              <DropdownItem onClick={() => {}}>Perfil</DropdownItem>
+              <DropdownItem onClick={
+                () => {
```

Comment:
Extract this `arrow function` from this render as an instance method. In general, avoid creating new references in renders.

Reply:
Complementary material: https://auth0.com/blog/optimizing-react/